src/mlpack/methods/ann/layer/cross_entropy_error.hpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file cross_entropy_error.hpp
+ * @author Konstantin Sidorov
+ *
+ * Definition of the cross-entropy performance function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_CROSS_ENTROPY_ERROR_HPP
+#define MLPACK_METHODS_ANN_LAYER_CROSS_ENTROPY_ERROR_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The cross-entropy performance function measures the network's
+ * performance according to the cross-entropy
+ * between the input and target distributions.
```

Comment:
I think you can reduce the number of lines here, it seems like they are nowhere near as long as 80 characters. :)  (I think this also applies elsewhere)